### PR TITLE
Refactor local scenario intent creation logic

### PR DESCRIPTION
**Pull request by creativedrewy**
This will make it easier for some of the future functionality to come.

### CODE REVIEW

**Review by olivierg13 — APPROVED** [review]

**Review by sdlaver — CHANGES_REQUESTED** [review]
Mostly good, just requesting minor changes

File: android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/scenario/LocalAssociationIntentCreator.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                        associationToken)
+                .appendQueryParameter(AssociationContract.LOCAL_PARAMETER_PORT,
+                        Integer.toString(port));
+
+        return new Intent()
+                .setAction(Intent.ACTION_VIEW)
+                .addCategory(Intent.CATEGORY_BROWSABLE)
+                .setData(dataUriBuilder.build());
+    }
+
```
[the remark this answers is not in the document]
Comment (creativedrewy) [reply android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/scenario/LocalAssociationIntentCreator.java:45]: Why static over instantiatable? I default to instances for DI purposes, though for this case I can definitely be talked out of it.

File: android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/scenario/LocalAssociationIntentCreator.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package com.solana.mobilewalletadapter.clientlib.scenario;
```
[the remark this answers is not in the document]
Comment (creativedrewy) [reply android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/scenario/LocalAssociationIntentCreator.java:1]: Is there a template for that?

File: android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/scenario/LocalAssociationIntentCreator.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import androidx.annotation.NonNull;
+
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterSession;
+import com.solana.mobilewalletadapter.common.AssociationContract;
+
+public class LocalAssociationIntentCreator {
+
+    @NonNull
+    public Intent createAssociationIntent(Uri endpointPrefix, int port, MobileWalletAdapterSession session) {
```
[the remark this answers is not in the document]
Comment (creativedrewy) [reply android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/scenario/LocalAssociationIntentCreator.java:15]: See my other comment about static/versus instance. Just wanna know your thoughts.

**Comment by creativedrewy** [comment]
All static now!

File: android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/scenario/LocalAssociationIntentCreator.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdlaver) [reply android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/scenario/LocalAssociationIntentCreator.java:51]: It avoids creating more work later for the GC. For a single class that is only instantiated very occasionally, it doesn't really matter, but just part of my general guide of good Java programming practices.

**Pull request merged**